docker/Dockerfile.devito

```diff
@@ -4,14 +4,38 @@
 
 # Base image with compilers
 ARG base=devitocodes/bases:cpu-gcc
+ARG petsc=""
+
+##################  Install PETSc ############################################
+FROM $base as petsc_builder 
+
+ARG petsc
+
+RUN if [ "$petsc" = "true" ]; then \
+        apt-get update && apt-get install -y libopenblas-serial-dev git pkgconf mpich && \
+        python3 -m venv /venv && \
+        /venv/bin/pip install --no-cache-dir --upgrade pip && \
+        /venv/bin/pip install --no-cache-dir --no-binary numpy numpy && \
```

Comment:
why?

Reply:
It is to avoid using pre-built binary wheels for numpy. This is important to ensure that PETSc and numpy are built against the same BLAS/LAPACK library. This is why the PETSc install stage is before Devito.